src/dox/Processor.hx

```diff
@@ -114,6 +97,138 @@ class Processor {
 		return newRoot;
 	}
 
+	function populateFieldsOfAbstract(theAbstract:Abstractdef, root:TypeRoot) {
+		if (hasDoxMetadata(theAbstract.impl.meta, "is-populated")) {
+			return; // nothing to do
+		}
+
+		var statics = new Array<ClassField>();
+		var fields = new Array<ClassField>();
+
+	 	// collect direct members
+		for (cf in theAbstract.impl.statics) {
+			switch (cf.type) {
+				// handling functions
+				case CFunction(args, _):
+					if (cf.name == "_new") { // constructor
+						fields.push(cf);
+					} else if (args.length == 0 || args[0].name != "this") {
```

Comment:
Instead of doing this, we should check for the presence of the `@:impl` metadata, which is how the compiler distinguishes these cases as well.

Reply:
That metadata is not present (maybe missing from the XML?). All there is for a constructor is:
```haxe
{
  name: _new, line: 34, 
  isFinal: false, 
  isPublic: true, 
  set: RMethod, platforms: [unused], 
  doc: * instance members, 
  params: [], 
  get: RNormal, 
  isOverride: false, 
  meta: [{name: :noCompletion, params: []}], 
  overloads: null, 
  type: CFunction([],CAbstract(dox.test.TestAbstract,[])), 
  expr: null
}
```

Reply:
Right, I forgot that I replaced that metadata with a class field flag in https://github.com/HaxeFoundation/haxe/pull/9747.  That likely means that #300 is a regression. In this case your approach is fine and should work for XML generated from lower Haxe version as well.